client/api/src/backend.rs

```diff
@@ -418,15 +418,28 @@ pub trait Backend<Block: BlockT>: AuxStore + Send + Sync {
 	) -> sp_blockchain::Result<()>;
 
 	/// Commit block insertion.
-	fn commit_operation(&self, transaction: Self::BlockImportOperation) -> sp_blockchain::Result<()>;
+	///
+	/// The `finalization_check` closure checks that if an imported block is able to be
+	/// finalized. If the check returns false, the import succeeds, but with the imported block
+	/// unfinalized.
+	fn commit_operation<F: Fn(&Block::Hash) -> bool>(
+		&self,
+		transaction: Self::BlockImportOperation,
+		finalization_check: &F,
+	) -> sp_blockchain::Result<()>;
 
 	/// Finalize block with given Id.
 	///
 	/// This should only be called if the parent of the given block has been finalized.
-	fn finalize_block(
+	///
+	/// The `finalization_check` closure checks that if an imported block is able to be
+	/// finalized. If the check returns false, the import succeeds, but with the imported block
+	/// unfinalized.
+	fn finalize_block<F: Fn(&Block::Hash) -> bool>(
```

Comment:
It seems like `finalization_check` should really be checked before calling this function. Also documentation mentions "import" which is not part of this function.